impc_etl/jobs/parse/gene_production_status_parser.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import textwrap
+from typing import Dict, List
+
+from airflow.sdk import Variable, asset
+from pyspark.sql.functions import (
+    when, col, lit, lower, udf, collect_set, array_contains
+)
+from pyspark.sql.types import StringType, IntegerType
+
+from impc_etl.utils import utils
+from impc_etl.utils.airflow import create_input_asset, create_output_asset
+from impc_etl.utils.spark import with_spark_session
+
+logger = logging.getLogger("airflow.task")
+
+dr_tag = Variable.get("data_release_tag")
+
+gentar_gene_interest_asset = create_input_asset("tracking/gene_interest.tsv")
+product_report_parquet_asset = create_input_asset("output/product_report_parquet")
+gene_status_output_asset = create_output_asset("gene_status_parquet")
+
+@asset.multi(
+    schedule=[gentar_gene_interest_asset, product_report_parquet_asset],
+    outlets=[gene_status_output_asset],
+    dag_id=f"{dr_tag}_gene_production_status_parser",
+    description=textwrap.dedent(
+        """
+        Parses gene production and phenotyping statuses by integrating gene interest data from Gentar
+        with product report outputs. This pipeline standardizes and maps various gene status fields 
+        (e.g., ES cell, CRISPR, conditional/null alleles, phenotyping) and saves the 
+        final gene status summary in a Parquet format. It supports data tracking for IMPC and MGI genes.
```

Comment:
Not clear what MGI genes means in this context, you can drop the last sentence, it doesn't add any information.